spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -198,8 +246,35 @@ private DispatchQueryResponse handleNonIndexQuery(DispatchQueryRequest dispatchQ
             tags,
             false,
             dataSourceMetadata.getResultIndex());
-    String jobId = emrServerlessClient.startJobRun(startJobRequest);
-    return new DispatchQueryResponse(jobId, false, dataSourceMetadata.getResultIndex());
+    if (sessionManager.isEnabled()) {
```

Comment:
Why only in handleNonIndexQuery?

Reply:
Got it Index Query doesn't need session
and ppl is handled via handleNonIndexQuery.